specification/streamanalytics/resource-manager/Microsoft.StreamAnalytics/preview/2020-03-01-preview/privateEndpoints.json

```diff
@@ -0,0 +1,358 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "StreamAnalyticsManagementClient",
+    "version": "2020-03-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
```

Comment:
Just a reference, some common def https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v1/privatelinks.json

Reply:
@weidongxu-microsoft  for the reason, share response from @atpham256 at private PR:

> Our PrivateEndpoint definition is different from the common one. We are trying to keep it as close as possible to Azure Network's definition but it is not completely the same.
> 
> Also, for the common definition, your private endpoint definition only contains and "id" property. We need a lot more than that. Customer needs to be able to send us the privateLinkServiceId and groupId, for example.
> 
> Another thing to note here is that we are not a resource like SQL or Storage where a user is connecting to our resource so our integration with private endpoints is not the same. I think Azure Network is using the term "transitive private endpoint access" for our scenario.

https://github.com/Azure/azure-rest-api-specs-pr/pull/1262#discussion_r432212512